tests/utils/profilehandler/data_preparation.go

```diff
@@ -0,0 +1,222 @@
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+)
+
+func PrepareVersion(client *rosacli.Client, versionRequirement string, channelGroup string, hcp bool) (
+	*rosacli.OpenShiftVersionTableOutput, error) {
+	versionList, err := client.Version.ListAndReflectVersions(channelGroup, hcp)
+	if err != nil {
+		return nil, err
+	}
+
+	if CON.VersionLatestPattern.MatchString(versionRequirement) {
+		return versionList.Latest()
+	} else if CON.VersionMajorMinorPattern.MatchString(versionRequirement) {
+		version, err := versionList.FindNearestBackwardMinorVersion(versionRequirement, 0, true)
+		return version, err
+	} else if CON.VersionRawPattern.MatchString(versionRequirement) {
+		return &rosacli.OpenShiftVersionTableOutput{
+			Version: versionRequirement,
+		}, nil
+	} else if CON.VersionFlexyPattern.MatchString(versionRequirement) {
+		return nil, nil // Implement later
+	}
+	return nil, fmt.Errorf("not supported version requirement: %s", versionRequirement)
+}
+
+// PrepareNames will generate the name for cluster creation
+// if longname is set, it will generate the long name with CON.DefaultLongClusterNamelength
+func PreparePrefix(profilePrefix string, nameLength int) string {
+	if nameLength > ocm.MaxClusterNameLength {
+		panic(fmt.Errorf("name length %d is longer than allowed max name length %d", nameLength, ocm.MaxClusterNameLength))
+	}
+	return common.GenerateRandomName(profilePrefix, nameLength-len(profilePrefix)-1)
+}
+
+func PrepareSubnetsDummy(vpcID string, region string, zones string) map[string][]string {
+	return map[string][]string{}
+}
+
+func PrepareProxysDummy(vpcID string, region string, zones string) map[string]string {
+	return map[string]string{
+		"https_proxy":  "",
+		"http_proxy":   "",
+		"no_proxy":     "",
+		"ca_file_path": "",
+	}
+}
+
+func PrepareKMSKeyDummy(region string) string {
+	return ""
+}
+
+func PrepareSecurityGroupsDummy(vpcID string, region string, securityGroupCount int) []string {
+	return []string{}
+}
+
+// PrepareAccountRoles will prepare account roles according to the parameters
+// openshiftVersion must follow 4.15.2-x format
+func PrepareAccountRoles(client *rosacli.Client,
+	namePrefix string,
+	hcp bool,
+	openshiftVersion string,
+	channelGroup string,
+	path string,
+	permissionsBoundary string) (
+	accRoles *rosacli.AccountRolesUnit, err error) {
+	flags := []string{
+		"--prefix", namePrefix,
+		"--mode", "auto",
+		"-y",
+	}
+	if openshiftVersion != "" {
+		majorVersion := common.SplitMajorVersion(openshiftVersion)
+		flags = append(flags, "--version", majorVersion)
+	}
+	if channelGroup != "" {
+		flags = append(flags, "--channel-group", channelGroup)
+	}
+	if hcp {
+		flags = append(flags, "--hosted-cp")
+	} else {
+		flags = append(flags, "--classic")
+	}
+	if path != "" {
+		flags = append(flags, "--path", path)
+	}
+	if permissionsBoundary != "" {
+		flags = append(flags, "--permissions-boundary", permissionsBoundary)
+	}
```

Comment:
NP : I would pull this into its own function, just to make this overall function more readable. And also easier to add a unit test if needed for prepairing the flags.

Reply:
Just to add, yeah might be easier to read, if we pull out prepairing the flags into their own funcs in this file, would clean up these prepare commands IMO

Reply:
Added a new function named `GenerateAccountRoleCreationFlag`